src/main/java/org/spongepowered/common/mixin/core/world/MixinExplosion.java

```diff
@@ -56,11 +56,21 @@
     @SuppressWarnings("rawtypes")
     @Shadow public List affectedBlockPositions;
 
+    private boolean shouldBreakBlocks;
```

Comment:
Change it so that the explosion builder is setting isSmoking and then sets the should break blocks correctly. no need for confusion.